src/coach/defence/DefencePositionCoach.cpp

```diff
@@ -173,24 +185,32 @@ DefenderBot DefencePositionCoach::createBlockOnLine(const Field &field, const Po
 DefenderBot DefencePositionCoach::createBlockPass(const Field &field, PossiblePass &pass, const Vector2 &blockPoint) {
     DefenderBot bot;
     bot.type = botType::BLOCKPASS;
-    bot.blockFromID = pass.toBot.id;
+    bot.blockFromID = pass.toBot->getId();
     bot.targetPos = blockPoint;
     bot.orientation = pass.faceLine();
     return bot;
 }
-std::shared_ptr<Vector2> DefencePositionCoach::blockOnDefenseAreaLine(const Field &field, const PossiblePass &pass, const world::WorldData &world) {
-    auto visibleParts = FieldComputations::getVisiblePartsOfGoal(field, true, pass.endPos, world);
+
+std::shared_ptr<Vector2> DefencePositionCoach::blockOnDefenseAreaLine(const Field &field, const PossiblePass &pass) {
+    std::vector<Line> visibleParts = simulatedVisibleGoalParts(field, pass);
+
     // get the largest segment
     std::sort(visibleParts.begin(), visibleParts.end(), [](const Line &a, const Line &b) { return abs(a.end.y - a.start.y) > abs(b.end.y - b.start.y); });
     if (!visibleParts.empty()) {
         return blockOnDefenseLine(field, visibleParts[0], pass.endPos);
     }
     return nullptr;
 }
+std::vector<Line> DefencePositionCoach::simulatedVisibleGoalParts(const Field &field, const PossiblePass &pass) const {// create a vector with all robots
+    std::vector<world_new::view::RobotView> allSimulatedRobots = simulatedRobotsUs;
+    allSimulatedRobots.insert(allSimulatedRobots.end(), simulatedRobotsThem.begin(), simulatedRobotsThem.end());
+    auto visibleParts = FieldComputations::getVisiblePartsOfGoalByObstacles(field, true, pass.endPos, allSimulatedRobots);
```

Comment:
This is probably too many robots, we tend to only consider our robots for obstacles.